src/System.Private.ServiceModel/src/System/ServiceModel/CommunicationException.cs

```diff
@@ -2,13 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Runtime.Serialization;
 
 namespace System.ServiceModel
 {
+    [Serializable]
     public class CommunicationException : Exception
     {
         public CommunicationException() { }
         public CommunicationException(string message) : base(message) { }
         public CommunicationException(string message, Exception innerException) : base(message, innerException) { }
+        protected CommunicationException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```

Comment:
`CommunicationException` is in S.SM.P contract. Should this ctor be added to contract?